cmake/OpenCVFindIPP.cmake

```diff
@@ -84,7 +84,7 @@ endmacro()
 # This macro uses IPP_ROOT_DIR variable
 # TODO Cleanup code after ICV package stabilization
 macro(ipp_detect_version)
-  set(IPP_INCLUDE_DIRS ${IPP_ROOT_DIR}/include)
+  get_filename_component(IPP_INCLUDE_DIRS ${IPP_VERSION_FILE} PATH)
```

Comment:
Check on the line 93 may fail and fail IPP detection completelly.

Reply:
https://github.com/opencv/opencv/blob/c577b2440643bf98942cfd8c49478c545c3fa56b/cmake/OpenCVFindIPP.cmake#L90-L97

> In OneAPI 2024, `inlcude/ipp.h` is the only one that has not moved to the directory `include/ipp/`. 

```
$ tree  /opt/intel/oneapi/ipp/2021.11/include/ -L 2
/opt/intel/oneapi/ipp/2021.11/include/
├── ipp
│   ├── ippbase.h
...
│   ├── ippvm.h
│   ├── iw
│   └── iw++
└── ipp.h

```

Reply:
Thank you for the information!

Perhaps we should not change `IPP_INCLUDE_DIRS` to the nested directory. Or add both locations.

I see these IPP includes in OpenCV:

```
./modules/core/include/opencv2/core/private.hpp:220:#include "ippversion.h"
./modules/core/include/opencv2/core/private.hpp:230:#include "ippicv.h"
./modules/core/include/opencv2/core/private.hpp:232:#include "ipp.h"
```

/cc @eplankin

Reply:
Well, top-level ipp.h is just a shim:

```
#include <ipp/ipp.h>
```